src/main/java/dan200/computercraft/shared/peripheral/speaker/SpeakerPeripheral.java

```diff
@@ -127,38 +125,37 @@ public String getType()
     }
 
     @Nonnull
-    private synchronized Object[] playNote(Object[] arguments, ILuaContext context) throws LuaException
+    private synchronized Object[] playNote( Object[] arguments, ILuaContext context ) throws LuaException
     {
-        double volume = 1f;
-        double pitch = 1f;
+        float volume = 1.0f;
+        float pitch = 1.0f;
 
         // Check if arguments are correct
-        if (arguments.length == 0) // Too few args
+        if( arguments.length == 0 ) // Too few args
         {
-            throw new LuaException("Expected string, number (optional), number (optional)");
+            throw new LuaException( "Expected string, number (optional), number (optional)" );
         }
 
-        if (!(arguments[0] instanceof String)) // Arg wrong type
+        if( !(arguments[0] instanceof String) ) // Arg wrong type
         {
             throw new LuaException("Expected string, number (optional), number (optional)");
         }
 
-        if (!SoundEvent.REGISTRY.containsKey(new ResourceLocation("block.note." + arguments[0])))
+        if ( !SoundEvent.REGISTRY.containsKey( new ResourceLocation( "block.note." + arguments[0] ) ) )
         {
             throw new LuaException("Invalid instrument, \"" + arguments[0] + "\"!");
         }
 
-        if (arguments.length > 1)
+        if ( arguments.length > 1 )
         {
-            if (!(arguments[1] instanceof Double) && arguments[1] != null)   // Arg wrong type
+            if ( !(arguments[1] instanceof Double) && arguments[1] != null )   // Arg wrong type
             {
-                throw new LuaException("Expected string, number (optional), number (optional)");
+                throw new LuaException( "Expected string, number (optional), number (optional)" );
             }
             volume = arguments[1] != null ? ((Double) arguments[1]).floatValue() : 1f;
-
         }
 
-        if (arguments.length > 2)
+        if( arguments.length > 2 )
         {
             if (!(arguments[1] instanceof Double) && arguments[2] != null)  // Arg wrong type
```

Comment:
I think this should say `(!(arguments[1] instanceof Double) && arguments[2] != null)`